src/rd.h

```
@@ -100,6 +100,15 @@
 /** Assert if reached */
 #define RD_NOTREACHED() rd_assert(!*"/* NOTREACHED */ violated")
 
+/** Assert if reached */
+#define RD_BUG(...) do {                                                \
```

nice! with that said, I've been wondering about the tradeoff between fail fast resulting in higher quality software and user perceptions (!!!! BUG is probably a worse perception than some subtly undefined behavior which may go unnoticed). i was always strongly on the side of fail fast, but i've started to wonder about it from a broader perspective.

---

If the user sees this we, as developers, have failed anyway.